dashboard/src/components/providers/ConfigProvider.tsx

```diff
@@ -0,0 +1,50 @@
+import { createContext, useContext, useEffect, useState } from 'react';
+
+import config from '@/config';
+import { ConfigEnvironments, ConfigInterface, throwError } from '@/config/config';
+
+interface AppConfig {
+  uid?: string;
+  telemetry?: 'on' | 'off';
+  iasqlEnv: ConfigEnvironments;
+  config: ConfigInterface;
+  configError?: string | null;
+}
+
+const AppConfigContext = createContext<AppConfig>({} as AppConfig);
+
+const useAppConfigContext = () => {
+  return useContext(AppConfigContext);
+};
+
+const AppConfigProvider = ({ children }: { children: any }) => {
+  const [appConfig, setAppConfig] = useState({} as AppConfig);
+  useEffect(() => {
+    const getConfig = async () => {
+      const response = await fetch(`/api/config`);
+      if (response.status === 200) {
+        const configJson = await response.json();
+        if (!Object.keys(appConfig).length) {
```

Comment:
Why this conditional?

Reply:
this method should be called just once. This is just a make sure to only set the values once and to avoid a re-rendering if for any reason it gets called again.